Client/Frontend/Fakespot/FakespotViewController.swift

```diff
@@ -6,13 +6,81 @@ import Common
 import ComponentLibrary
 import UIKit
 import Shared
+import Redux
+
+struct FakespotState: ScreenState, Equatable {
+    var isSettingsExpanded: Bool
+    var isReviewQualityExpanded: Bool
+    let logger: Logger = DefaultLogger.shared
+
+    enum Action: Redux.Action {
```

Comment:
This is one thing that I like to do keeping related things together. State and actions seems to always be tight with a feature so having them like separate structs in separate files makes us jump in between and look for what we need. I like this structure better:

```swift
struct Store<State, Action> {
    struct State {
    }

    enum Action {
    }

    static let reducer: <State, Action> ...
}

Reply:
I like this structure too, we could start using it

Reply:
State and Action aren't necessarily linked, you can have many actions that result in triggering a change to the same state object. I might be misunderstanding the suggestion tho.

Reply:
```swift
struct Feature<State, Action> {
    struct State {
    }

    enum Action {
    }

    static let reducer: <State, Action> ...
}

Yes, coming back to this example I changed the structure to be a `Feature` instead of a `Store`. My understanding of Redux is that you have the feature stare and feature actions. Actions are send to the reducer to change the feature state, and stat local action state is derived from the app state. So actions imo are tight to the feature state.

I would find it a bit confusing for example if we get to TabFeature and have to handle a ThemeAction.

Reply:
Ah, I mistook it to mean one action per one state. Yes this improvement seems fine with me.